### ISSUE TITLE
WCAG 2.2: Investigate the use of focus styles on large target areas

### ISSUE THREAD

**Issue by querkmachine** [alphagov/govuk-frontend#3839]
## What

Some components—notably the accordion and step-by-step navigation—include large clickable regions to expand and collapse each section. 

Currently, these areas turn light grey on hover.

![Screenshot of the accordion component with the first section heading hovered, giving it a light grey background that extends across the entire heading.](https://github.com/alphagov/govuk-frontend/assets/1253214/ad23361a-0e02-4f0e-aa06-221a2cab5f57)

However, the focus style used on these elements does not cover the same area.

![Screenshot of the accordion component with the first section heading focused, giving only the text on the left of the heading a yellow background with a black underline.](https://github.com/alphagov/govuk-frontend/assets/1253214/0508ebea-b29a-4c39-8b5c-af5afdb517d8)

## Why

The WCAG 2.2 Level AAA ['Focus Appearance'](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html) criterion states:

> When the keyboard [focus indicator](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html#dfn-focus-indicator) is visible, an area of the focus indicator meets all the following:
> 
> - is at least as large as the area of a 2 [CSS pixel](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html#dfn-css-pixel) thick [perimeter](https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance.html#dfn-perimeter) of the unfocused component or sub-component, and
> - has a contrast ratio of at least 3:1 between the same pixels in the focused and unfocused states.

The focus style not encompassing the entire clickable area may constitute a failure under this criterion.

Additionally, the current styles may fail the Level AA ['Focus Not Obscured (Minimum)'](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-minimum.html) and Level AAA ['Focus Not Obscured (Enhanced)'](https://www.w3.org/WAI/WCAG22/Understanding/focus-not-obscured-enhanced) criteria when used in combination with certain assistive technologies, such as screen magnifiers, as the focus-styled elements may not be visible on the user's screen even if other parts of the clickable region are.

## Assumptions

- A screen magnification user may not see the focus style if their magnifier is centred to the right of the heading, causing the text and focus style to not be visible on screen.

## Timebox

3 days

We should review progress after this period of time has elapsed, even [if the spike has not been 'completed'](https://docs.google.com/document/d/17W_d0aYszgh7HTgnTHZLxXDu_TJyu8zzWt_lXH8VaHA/edit#heading=h.jd56itytegn6)

## Who is working on this?

Spike lead: ?

Spike buddy: ?

## Questions to answer

- [ ] Is the current focus style (text focus only) adequate to meet 'Focus Not Obscured (Minimum)'? 
- [ ] Is the current focus style (text focus only) adequate to meet the Level AAA criterion ('Focus Appearance' and 'Focus Not Obscured (Enhanced)')? 
- [ ] Is the alternative focus style (entire link) adequate to meet 'Focus Not Obscured (Minimum)'? 
- [ ] Is the alternative focus style (entire link) adequate to meet the Level AAA criterion ('Focus Appearance' and 'Focus Not Obscured (Enhanced)')? 

## Done when

You may find it helpful to refer to our [expected outcomes of spikes](https://docs.google.com/document/d/17W_d0aYszgh7HTgnTHZLxXDu_TJyu8zzWt_lXH8VaHA/edit#heading=h.mmkqzigd11rs).

- [ ] Questions have been answered or we have a clearer idea of how to get to our goal
- [ ] Findings have been reviewed and agreed with at least one other person
- [ ] Findings have been shared, e.g: via a write-up on the ticket, at a show & tell or team meeting
- [ ] A decision has been made on whether we need to amend the focus style used by the components
- [ ] If necessary, an issue to amend the focus style has been opened

**Comment by 36degrees** [comment]
Relevant discussion in the WCAG repo here: https://github.com/w3c/wcag/issues/1337

**Comment by querkmachine** [comment]
As perhaps some other assumptions to test, I have a concern that there is no easy answer to this:

Our current (assumed) probem is that the current focus style does not highlight the entire area, so a magnification tool user whose viewport isn't near the text could not tell that the link they are looking at is focused.

Conversely, however, our focus style depends upon the combination of the black bottom border and the yellow background—as the yellow background alone does not provide sufficient contrast when used on white. Even if we were to make the entire heading take on the focus style, the lack of the 'underline' doesn't mean the focus style will necessarily be visible to someone with low vision.

**Comment by CharlotteDowns** [comment]
## Design prototyping

I've been [prototyping some thoughts on what an updated focus state could look like in this google slide deck](https://docs.google.com/presentation/d/1SJu4zZoXQdcCZjas6dB9eYI7Nb6b2n3P55EhB9DOZzQ/edit?pli=1#slide=id.g25db8cd05e6_0_12).

**Comment by CharlotteDowns** [comment]
This has prompted a [review and rationalisation of our approach to styling focus states govuk-design-system](https://github.com/alphagov/govuk-design-system/issues/3007). I'm closing this comment in favour of moving the conversations to the govuk-design-system/3007.

For the moment I think we are agreed this is a stretch goal of the WCAG 2.2 related updates and actions.

**Comment by dav-idc** [comment]
During my [WCAG 2.2 'complex testing' epic](https://github.com/alphagov/govuk-frontend/issues/4347) work, I've discovered that our lack of a conclusive image focus style affects our focus state for the crest in the footer. Just wanted to make a note here.

Our current focus state for the crest in the footer relies on areas of transparency in the image. And the overall effect is very low contrast between the yellow focus background and the grey crest.
![The grey footer crest with a yellow focus applied.](https://github.com/alphagov/govuk-frontend/assets/98318778/7723143c-fc67-46fa-8e43-115a1b29032c)

I've made a note in my [testing spreadsheet](https://docs.google.com/spreadsheets/d/1PpKvhCGZJb1j0ZUUY2ZuA0CBQeu04yFD1bumiL8wGX8/edit?usp=sharing) that this image focus state work interacts with the testing of the footer crest focus appearance.

It looks like the crest focus appearance will need to wait for the work to start on the ['Rationalise our approach to styling focus states' ticket](https://github.com/alphagov/govuk-design-system/issues/3007).

**Comment by CharlotteDowns** [comment]
I’m happy to look into this on Monday.

**Comment by CharlotteDowns** [comment]
## Footer image link review and suggestions

Until we kick off ['Rationalise our approach to styling focus states' ticket](https://github.com/alphagov/govuk-design-system/issues/3007), I suggest surrounding the footer image with a similar black border as is used in the `@mixin govuk-focused-box` used for the [linked image focus style](https://github.com/alphagov/govuk-frontend/pull/3819/files).

<img width="1425" alt="The grey footer crest with a yellow focus applied and surround black border" src="https://github.com/alphagov/govuk-frontend/assets/56260216/79914be8-6342-457d-b55e-580424a22daf">

We defined the footer image link style as a ['logo focus style' in the grouping focus states team workshop (2021)](https://padlet.com/charlottedowns/grouping-focus-states-5apruycsoww8cmth). We suggested the following principles/pieces of guidance/rules for focus states for logos:

> Use a combination of yellow and black, to ensure contrast with both light and dark backgrounds. Don't try to style the borders of complex shapes like logos.
